src/libs/actions/App.js

```diff
@@ -48,15 +47,24 @@ Onyx.connect({
     },
 });
 
-const allPolicies = {};
+let policyIDList = [];
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
-    callback: (val, key) => {
-        if (!val || !key) {
+    waitForCollectionCallback: true,
+    callback: (val) => {
```

Comment:
I still think this code is kind of overkill for what it's doing. Does this work? It seems more simple:

```
let policyIDList = [];
Onyx.connect({
    key: ONYXKEYS.COLLECTION.POLICY,
    waitForCollectionCallback: true,
    callback: (policies) => {
        policyIDList = _.pluck(policies, 'id');
    },
});
```

Reply:
It does! But I found a bug with `waitForCollectionCallback` param while testing this solution. See https://github.com/Expensify/react-native-onyx/pull/167 for details

Reply:
I updated Onyx and the logic in the connect call! It's much simpler now, thanks for the thorough review!